wasm/src/algorithms/poseidon/mod.rs

```diff
@@ -56,31 +57,31 @@ mod tests {
             let native_field_array = create_native_field_vector(Some(*count));
 
             // Hash the field array using all Poseidon hasher instances.
-            let hash_2 = poseidon2.hash(js_array_from_fields(&native_field_array)).unwrap();
-            let hash_4 = poseidon4.hash(js_array_from_fields(&native_field_array)).unwrap();
-            let hash_8 = poseidon8.hash(js_array_from_fields(&native_field_array)).unwrap();
-            let hash_2_scalar = poseidon2.hash_to_scalar(js_array_from_fields(&native_field_array)).unwrap();
-            let hash_4_scalar = poseidon4.hash_to_scalar(js_array_from_fields(&native_field_array)).unwrap();
-            let hash_8_scalar = poseidon8.hash_to_scalar(js_array_from_fields(&native_field_array)).unwrap();
-            let hash_2_group = poseidon2.hash_to_group(js_array_from_fields(&native_field_array)).unwrap();
-            let hash_4_group = poseidon4.hash_to_group(js_array_from_fields(&native_field_array)).unwrap();
-            let hash_8_group = poseidon8.hash_to_group(js_array_from_fields(&native_field_array)).unwrap();
+            let hash_2 = poseidon2.hash(js_array_from_fields!(&native_field_array)).unwrap();
+            let hash_4 = poseidon4.hash(js_array_from_fields!(&native_field_array)).unwrap();
+            let hash_8 = poseidon8.hash(js_array_from_fields!(&native_field_array)).unwrap();
+            let hash_2_scalar = poseidon2.hash_to_scalar(js_array_from_fields!(&native_field_array)).unwrap();
+            let hash_4_scalar = poseidon4.hash_to_scalar(js_array_from_fields!(&native_field_array)).unwrap();
+            let hash_8_scalar = poseidon8.hash_to_scalar(js_array_from_fields!(&native_field_array)).unwrap();
+            let hash_2_group = poseidon2.hash_to_group(js_array_from_fields!(&native_field_array)).unwrap();
+            let hash_4_group = poseidon4.hash_to_group(js_array_from_fields!(&native_field_array)).unwrap();
+            let hash_8_group = poseidon8.hash_to_group(js_array_from_fields!(&native_field_array)).unwrap();
```

Comment:
Could you add an extra `wasm_bindgen_test` 

That:
1. For all the literal types that go from literal.to_plaintext().to_fields(), hashes them and then checks against the native implementation?
2. For a struct type that includes all the plaintext literal types we currently support, does the same as above?

Could you to the same for `BHP` and `Pedersen`? (Of course going from `literal.to_plaintext().to_bits()` instead)